WordPress/Classes/ViewRelated/Menus/MenusSelectionView.m

```diff
@@ -266,4 +281,46 @@ - (void)selectionItemObjectWasUpdatedNotification:(NSNotification *)notification
     }
 }
 
+#pragma mark - accessibility
+
+- (void)prepareForVoiceOver
+{
+    if ([self.selectedItem isMenuLocation]) {
+        [self configureLocationAccessibility];
+    } else if ([self.selectedItem isMenu] && self.selectedItemLocation != nil) {
+        [self configureMenuAccessibility];
+    }
+    NSString *expandedLocalizedString = NSLocalizedString(@"Expanded", @"Screen reader text to represent the expanded state of a UI control");
+    self.detailView.accessibilityValue = self.selectionItemsExpanded ? expandedLocalizedString : nil;
+    self.addNewItemView.accessibilityLabel = NSLocalizedString(@"Add new menu", @"Screen reader text for Menus button that adds a new menu to a site.");
+
+    [self configureAccessibilityGroups];
+}
+
+- (void)configureLocationAccessibility
+{
+    // Menu area: Header. 3 menu areas in this theme. Button. [hint] Expands to select a different menu area.
+    NSString *format = NSLocalizedString(@"Menu area: %@, %@", @"Screen reader string too choose a menu area to edit. First %@ is the name of the menu area (Primary, Footer, etc...). Second %@ is an already localized string saying the number of areas in this theme.");
+    self.detailView.accessibilityLabel = [NSString stringWithFormat:format,
+                                          self.selectedItem.displayName,
+                                          self.detailView.subTitleLabel.text];
+    self.detailView.accessibilityHint = NSLocalizedString(@"Expands to select a different menu area", @"Screen reader hint (non-imperative) about what does the site menu area selector button do.");
+}
+
+- (void)configureMenuAccessibility
+{
+    // Menu in area Header: Primary. 3 menus available. Button. [hint] Expands to select a different menu to edit.
+    NSString *format = NSLocalizedString(@"Menu in area %@: %@, %@", @"Screen reader string too choose a menu to edit. First %@ is the name of the menu area (Primary, Footer, etc...). Second %@ is name of the menu currently selected. Third is an already localized string saying the number of menus available in the menu area selected.");
+    self.detailView.accessibilityLabel = [NSString stringWithFormat:format,
+                                          self.selectedItemLocation.displayName,
```

Comment:
True that two `MenuSelectionitem`s to `MenusSelectionView` might be confusing, and it's used just on the `menus picker` use case.
Delegates are usually used to inform about stuff rather than requesting data, right? It would be better as a Data Source.

Since all the information is in the controller, I'll give it a try to set these labels in the controller itself, rather than passing the data to the views to set their own labels. Wdyt?

Reply:
> Delegates are usually used to inform about stuff rather than requesting data, right? It would be better as a Data Source.
I was just referring to the delegate pattern (I didn't make that clear), i.e. the use of a protocol to fetch stuff  — which covers "delegates" or "data sources".

> Since all the information is in the controller, I'll give it a try to set these labels in the controller itself, rather than passing the data to the views to set their own labels. Wdyt?

I think that would also work 👍